src/cargo/ops/registry/info/mod.rs

```diff
@@ -0,0 +1,285 @@
+    let (mut package_id, is_member) = find_pkgid_in_ws(nearest_package, ws.as_ref(), spec);
+    let (use_package_source_id, source_ids) =
+        get_source_id_with_package_id(gctx, package_id, reg_or_index.as_ref())?;
+    // If we don't use the package's source, we need to query the package ID from the specified registry.
+    if !use_package_source_id {
+        package_id = None;
+    }
+
+    let msrv_from_nearest_manifest_path_or_ws =
+        try_get_msrv_from_nearest_manifest_or_ws(nearest_package, ws.as_ref());
+    // If the workspace does not have a specific Rust version,
+    // or if the command is not called within the workspace, then fallback to the global Rust version.
+    let rustc_version = match msrv_from_nearest_manifest_path_or_ws {
+        Some(msrv) => msrv,
+        None => {
+            let current_rustc = gctx.load_global_rustc(ws.as_ref())?.version;
+            // Remove any pre-release identifiers for easier comparison.
+            // Otherwise, the MSRV check will fail if the current Rust version is a nightly or beta version.
+            semver::Version::new(
+                current_rustc.major,
+                current_rustc.minor,
+                current_rustc.patch,
+            )
+            .into()
+        }
+    };
+    // Only suggest cargo tree command when the package is not a workspace member.
+    // For workspace members, `cargo tree --package <SPEC> --invert` is useless. It only prints itself.
+    let suggest_cargo_tree_command = package_id.is_some() && !is_member;
+
+    let summaries = query_summaries(spec, &mut registry, &source_ids)?;
+    let package_id = match package_id {
+        Some(id) => id,
+        None => find_pkgid_in_summaries(&summaries, spec, &rustc_version, &source_ids)?,
+    };
+
+    let package = registry.get(&[package_id])?;
+    let package = package.get_one(package_id)?;
+    let owners = try_list_owners(
+        gctx,
+        &source_ids,
+        reg_or_index.as_ref(),
+        package_id.name().as_str(),
+    )?;
+    pretty_view(
+        package,
+        &summaries,
+        &owners,
+        suggest_cargo_tree_command,
+        gctx,
+    )?;
+
+    Ok(())
+}
+
+fn find_pkgid_in_ws(
+    nearest_package: Option<&Package>,
+    ws: Option<&Workspace<'_>>,
+    spec: &PackageIdSpec,
+) -> (Option<PackageId>, bool) {
+    let Some(ws) = ws else {
+        return (None, false);
+    };
+
+    if let Some(member) = ws.members().find(|p| spec.matches(p.package_id())) {
+        return (Some(member.package_id()), true);
+    }
+
+    let Ok((_, resolve)) = resolve_ws(ws, false) else {
+        return (None, false);
+    };
+
+    if let Some(package_id) = nearest_package
+        .map(|p| p.package_id())
+        .into_iter()
+        .flat_map(|p| resolve.deps(p))
+        .map(|(p, _)| p)
+        .filter(|&p| spec.matches(p))
+        .max_by_key(|&p| p.version())
+    {
+        return (Some(package_id), false);
+    }
+
+    if let Some(package_id) = ws
```

Comment:
I'm not sure if we should mention this detail here. From the start, I provided all the details about how we select a package, but Ed thought it was too detailed. He believes it might create more work if we want to change these implementation details.